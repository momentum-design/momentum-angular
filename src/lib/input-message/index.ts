export * from './input-message.component';
export * from './input-message.module';
