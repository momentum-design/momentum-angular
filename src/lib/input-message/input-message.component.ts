import { Component, Input } from '@angular/core';

let nextUniqueId = 0;

@Component({
  selector: 'md-input-message',
  template: `<ng-content></ng-content>`,
  styles: [],
  host: {
    'class': 'md-input__message',
    'role': 'alert',
    '[attr.id]': 'id'
  }
})

export class InputMessageComponent {
  /** @prop Unique HTML ID used for automated testing | null */
  @Input() id: string = `md-error-${nextUniqueId++}`;
}
