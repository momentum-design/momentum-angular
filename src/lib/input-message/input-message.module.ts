import { CommonModule } from '@angular/common';
import { NgModule, ModuleWithProviders } from '@angular/core';
import { InputMessageComponent } from './input-message.component';

@NgModule({
  declarations: [InputMessageComponent],
  imports: [CommonModule],
  exports: [InputMessageComponent],
})

export class InputMessageModule {
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: InputMessageModule,
      providers: [],
    };
  }
}
