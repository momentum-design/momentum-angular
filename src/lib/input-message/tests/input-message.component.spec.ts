import {
  async,
  ComponentFixture,
  TestBed
} from '@angular/core/testing';
import { InputMessageComponent } from '../input-message.component';

describe('InputMessageComponent', () => {
  let component: InputMessageComponent;
  let fixture: ComponentFixture<InputMessageComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [InputMessageComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(InputMessageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create and match snapshot', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should overwrite id', () => {
    expect(fixture.nativeElement.getAttribute('id')).toContain('md-error-');
    component.id = 'test-id';
    fixture.detectChanges();

    expect(fixture.nativeElement.getAttribute('id')).toContain('test-id');
  });
});
