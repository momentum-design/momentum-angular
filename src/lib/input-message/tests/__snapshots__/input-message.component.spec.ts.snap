// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputMessageComponent should create and match snapshot 1`] = `
<md-input-message
  id={[Function String]}
>
  
</md-input-message>
`;
